go-controller/pkg/node/helper_linux.go

```diff
@@ -61,6 +51,44 @@ func getDefaultGatewayInterfaceDetails() (string, []net.IP, error) {
 	return intfName, gatewayIPs, nil
 }
 
+func getDefaultGatewayInterfaceByFamily(family int) (string, net.IP, error) {
+	// filter the default route to obtain the gateway
+	filter := &netlink.Route{Dst: nil}
+	routes, err := netlink.RouteListFiltered(family, filter, netlink.RT_FILTER_DST)
+	if err != nil {
+		return "", nil, errors.Wrapf(err, "failed to get routing table in node")
+	}
+	// only support ECMP
+	if len(routes) != 1 {
```

Comment:
@aojea I'm a bit worried about this bit `len(routes) != 1`. NetworkManager uses multiple default routes by default, if you have multiple interfaces connected that *could* have the default route. It will then set metrics on those routes to make sure routing is deterministic, and to reduce connection downtime when one interface fails/unplugged/etc.

eg, if I have both wifi and ethernet connected, I will see:

```
$ ip route
default via 192.168.1.1 dev enp0s31f6 proto dhcp metric 100 
default via 192.168.1.1 dev wlp61s0 proto dhcp metric 600 
```

Would it instead make sense to check for multipath first, and if there is no multipath, fall back to using the first non-ECMP default route we find?

Reply:
absolutely, let me modify it